doc/contributing.rst

```diff
@@ -379,10 +379,31 @@ with ``git commit --no-verify``.
 Backwards Compatibility
 ~~~~~~~~~~~~~~~~~~~~~~~
 
-Please try to maintain backward compatibility. *xarray* has growing number of users with
+Please try to maintain backwards compatibility. *xarray* has a growing number of users with
 lots of existing code, so don't break it if at all possible.  If you think breakage is
-required, clearly state why as part of the pull request.  Also, be careful when changing
-method signatures and add deprecation warnings where needed.
+required, clearly state why as part of the pull request.
+
+Be especially careful when changing function and method signatures, because any change
+may require a deprecation warning. For example, if your pull request means that the
+argument ``old_arg`` to ``func`` is no longer valid, instead of simply raising an error if
+a user passes ``old_arg``, we would instead catch it:
+
+    def func(new_arg, old_arg=None):
+        if old_arg is not None:
+            from warnings import warn
+
+            warn(
+                "`old_arg` has been deprecated, and in the future will raise an error."
+                "Please use `new_arg` from now on."
+            )
```

Comment:
shouldn't we also specify the warning? I'd say `DeprecationWarning`, but the docs are not really clear on this: it says the target audience for `FutureWarning` is the users of an application, which I assume is not what we want.

Reply:
I've synthesized the difference as `FutureWarning` is for behavior changes, and `DeprecationWarning` is for removal. I'm still not confident that's exactly right! 

But `DeprecationWarning` is right here I think

Reply:
> I'm still not confident that's exactly right!

according to the [docs](https://docs.python.org/3/library/warnings.html#warning-categories) (immediately below the table), that was the case before `python=3.7`.